govc/flags/cluster.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flags
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/property"
+	"github.com/vmware/govmomi/view"
+	"github.com/vmware/govmomi/vim25/mo"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+type ClusterFlag struct {
+	common
+
+	*DatacenterFlag
+
+	Name string
+
+	cluster *object.ClusterComputeResource
+	pc      *property.Collector
+}
+
+var clusterFlagKey = flagKey("cluster")
+
+func NewClusterFlag(ctx context.Context) (*ClusterFlag, context.Context) {
+	if v := ctx.Value(clusterFlagKey); v != nil {
+		return v.(*ClusterFlag), ctx
+	}
+
+	v := &ClusterFlag{}
+	v.DatacenterFlag, ctx = NewDatacenterFlag(ctx)
+	ctx = context.WithValue(ctx, clusterFlagKey, v)
+	return v, ctx
+}
+
+func (f *ClusterFlag) Register(ctx context.Context, fs *flag.FlagSet) {
+	f.RegisterOnce(func() {
+		f.DatacenterFlag.Register(ctx, fs)
+
+		env := "GOVC_CLUSTER"
```

Comment:
define it as a const here: `govc/flags/client.go`?

Reply:
The client const's are referenced in more than 1 func.  GOVC_CLUSTER is only referenced within this func, you'll see the same pattern in other flags:
```console
% git --no-pager grep -n -e env\ \:\=\ \"GOVC_ --
govc/flags/datacenter.go:61:		env := "GOVC_DATACENTER"
govc/flags/datastore.go:63:		env := "GOVC_DATASTORE"
govc/flags/debug.go:51:		env := "GOVC_DEBUG"
govc/flags/folder.go:54:		env := "GOVC_FOLDER"
govc/flags/host_system.go:61:		env := "GOVC_HOST"
govc/flags/network.go:58:		env := "GOVC_NETWORK"
govc/flags/resource_pool.go:54:		env := "GOVC_RESOURCE_POOL"
govc/flags/storage_pod.go:39:		env := "GOVC_DATASTORE_CLUSTER"
govc/flags/virtual_app.go:57:		env := "GOVC_VAPP"
govc/flags/virtual_machine.go:60:		env := "GOVC_VM"
govc/vm/guest/auth.go:54:	env := "GOVC_GUEST_LOGIN"
```